### DISCUSSION TITLE
[Core Protocol] What are the security guarantees of L1Block?

### DISCUSSION THREAD

**Discussion by Wiezzel** [ethereum-optimism/developers#666] [Core Protocol] [answered]
### Did you check the documentation and specs?

- [X] Yes

### Was there anything unclear or missing?

I understand that `L1Block` is a pre-deployed contract on Optimism that is updated at the beginning of a new epoch to contain the most recent L1 block information. My question is: what are the security guarantees, or to put ut in another way, the trust assumptions of this information? Is it part of the consensus? Under which circumstances is it possible that an invalid L1 block information appears there?

### Did you check for duplicate questions?

- [X] Yes

### Please select the type of request

- [ ] Bug Report
- [ ] Feature Request

**Comment by hash1go** [comment, marked as answer]
Hello.
The security of the `L1Block` contract is based on the liveness of the sequencer node and trust in the L1 node that the sequencer node references. Typically, sequencer nodes operate their own L1 nodes.
The correctness of the `L1Block` contract can be verified by operating your own L1 node.

**Reply by Wiezzel** [reply]
But is it eventually part of the consensus? Can invalid hash in `L1Block` be challenged?

**Reply by hash1go** [reply]
Since OP Mainnet is a rollup, it does not have its own consensus. The transaction data posted to Ethereum becomes the canonical data.

> Can invalid hash in L1Block be challenged?

No, it cannot be challenged.

**Reply by Wiezzel** [reply]
@hash1go Thank you for the clarification. Do you perhaps know of any other way, how could a trusted (i.e. potentially challengeable) L1 hash be obtained on Optimism?

**Reply by hash1go** [reply]
The information obtained from `L1Block` can be cross-checked for consistency with data obtained by independently operating your own L1 node.

**Reply by Wiezzel** [reply]
Understood. This however doesn't meet my needs, as I am looking for a way verify this on-chain.

**Reply by hash1go** [reply]
If "on-chain" refers to the L2 environment, you cannot achieve what you're requesting. Trust that the "L1Block" contract is correct.

OP Stack is Ethereum rollup, and while Ethereum has the ability to challenge L2's state, OP Stack does not have the ability to challenge Ethereum's state.
